src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -603,17 +605,28 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                     CurlHandler.EventSourceTrace("Size: {0}", size, easy: easy);
                     try
                     {
-                        // The callback is invoked once per header; multi-line headers get merged into a single line.
-                        string responseHeader = Marshal.PtrToStringAnsi(buffer).Trim();
                         HttpResponseMessage response = easy._responseMessage;
 
-                        if (!TryParseStatusLine(response, responseHeader, easy))
+                        CurlResponseHeaderReader reader = new CurlResponseHeaderReader(buffer, size);
```

Comment:
Which makes me wonder: should [`HttpRuleParser.DefaultHttpEncoding`](https://github.com/dotnet/corefx/blob/11efc0d43d56e1c280653934b27ae5c1ef05af8c/src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs#L42-L47) be used to materialize the header value instead of `Marshal.PtrToStringAnsi`?